### PR TITLE
Fix DQN, C51, IQN loss functions to registry

### PR DESCRIPTION
**Pull request by jinPrelude**

### CODE REVIEW

File: rl_algorithms/dqn/agent.py [hunk 1]
```diff
@@ -135,6 +134,7 @@ def _init_network(self):
 
         self.dqn = BaseNetwork(self.backbone_cfg, self.head_cfg).to(device)
         self.dqn_target = BaseNetwork(self.backbone_cfg, self.head_cfg).to(device)
+        self._get_dqn_loss = build_loss(self.hyper_params.loss_type)
```
Comment (MrSyee) [inline rl_algorithms/dqn/agent.py:137]: should change the variable name: `self.loss_fn`

File: rl_algorithms/dqn/losses.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
```
Comment (MrSyee) [inline rl_algorithms/dqn/losses.py:1]: should add or modify description

File: rl_algorithms/dqn/agent.py (hunk 1 above)
Comment (jinPrelude) [reply rl_algorithms/dqn/agent.py]: resolved at 2113f91!

File: rl_algorithms/dqn/losses.py (hunk 2 above)
Comment (jinPrelude) [reply rl_algorithms/dqn/losses.py:1]: resolved at 2113f91!
